hooks/useLogin.ts

```diff
@@ -70,54 +71,44 @@ export function useLogin(config: LoginProps = {}) {
           getUserDetails(token),
         ])
 
-        if ('error' in details || details instanceof LocalError) {
-          $setStatus(STATUS.FAILED)
+        if (details instanceof LocalError) {
           // this is a visible error and a breaking error
+          $setStatus(STATUS.FAILED)
           $setError(details)
-          Promise.reject(details)
           return
         }
+
         if (details.statusCode && details.statusCode === 401) {
           $setStatus(STATUS.NOT_FOUND)
           $setError(new LocalError('No user found.', NO_MAGIC_USER))
           return
         }
+
         $setStatus(STATUS.LOADED)
         $setMetadata(details)
         $setMagicMetadata(magicMd)
-      } catch (err) {
-        if (err.toString().indexOf('-32603') > -1) {
+      } catch (err: unknown) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        if ((err as any)?.toString()?.indexOf('-32603') > -1) {
           $setStatus(STATUS.NOT_FOUND)
           return
         }
+
         throw err
       }
     }
+
     if (!$metadata) {
-      try {
-        checkLoggedIn()
-      } catch (e) {
+      checkLoggedIn().catch(e => {
```

Comment:
If you don't await this, then the try/catch doesn't do anything once the first async method starts.